Documentation/gettingstarted/local-redirect-policy.rst

```diff
@@ -301,5 +309,169 @@ configuring the CiliumLocalRedirectPolicy.
+             <https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/#configuration>`_
+             to fill in the required template variables ``__PILLAR__LOCAL__DNS__``, ``__PILLAR__DNS__DOMAIN__``,
+             and ``__PILLAR__DNS__SERVER__`` before applying the yaml.
+
+        .. parsed-literal::
+
+            $ wget \ |SCM_WEB|\/examples/kubernetes-local-redirect/node-local-dns.yaml
+
+            $ kubedns=$(kubectl get svc kube-dns -n kube-system -o jsonpath={.spec.clusterIP}) && sed -i "s/__PILLAR__DNS__SERVER__/$kubedns/g;" node-local-dns.yaml
+
+            $ kubectl apply -f node-local-dns.yaml
+
+    .. group-tab:: Manual Configuration
+
+         * Follow the official `NodeLocal DNSCache Configuration
+           <https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/#configuration>`_
+           to fill in the required template variables ``__PILLAR__LOCAL__DNS__``, ``__PILLAR__DNS__DOMAIN__``,
+           and ``__PILLAR__DNS__SERVER__`` before applying the yaml.
+
+         * Make sure to use a Node-local DNS image with a release version >= 1.15.16.
+           This is to ensure that we have a knob to disable dummy network interface creation/deletion in
+           Node-local DNS when we deploy it in non-host namespace.
+
+         * Modify Node-local DNS cache's deployment yaml to pass these additional arguments to node-cache:
+           ``-skipteardown=true``, ``-setupinterface=false``, and ``-setupiptables=false``.
+
+         * Modify Node-local DNS cache's deployment yaml to put it in non-host namespace by setting
+           ``hostNetwork: false`` for the daemonset.
+
+         * In the Corefile, bind to ``0.0.0.0`` instead of the static IP.
+
+         * In the Corefile, let CoreDNS serve health-check on its own IP instead of the static IP by
+           removing the host IP string after health plugin.
+
+         * Modify Node-local DNS cache's deployment yaml to point readiness probe to its own IP by
+           removing the ``host`` field under ``readinessProbe``.
+
+* Deploy local redirect policy (LRP) to steer DNS traffic to the node local dns cache.
+
+  .. parsed-literal::
+
+      $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes-local-redirect/node-local-dns-lrp.yaml
+
+  .. note::
+
+      * The LRP above uses ``kube-dns`` for the cluster DNS service, however if your cluster DNS service is different,
+        you will need to modify this example LRP to specify it.
+      * The namespace specified in the LRP above is set to the same namespace as the cluster's dns service.
+      * The LRP above uses the same port names ``dns`` and ``dns-tcp`` as the example quick deployment yaml, you will
+        need to modify those to match your deployment if they are different.
+
+After all ``node-local-dns`` pods are in ready status, DNS traffic will now go to the local node-cache first.
+You can verify by checking the DNS cache's metrics such as ``coredns_cache_hits_total`` and
```

Comment:
I was able to successfully follow the quick guide. However, there is a minor issue with the verification step. I don't see the `coredns_cache_hits_total` counter.  

Here are some of the metrics that were getting incremented on my cluster as I initiated DNS queries from one of the application pods - 

```
curl 10.16.153.17:9253/metrics | grep coredns_cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0# HELP coredns_cache_misses_total The count of cache misses.
# TYPE coredns_cache_misses_total counter
coredns_cache_misses_total{server="dns://0.0.0.0:53"} 10
# HELP coredns_cache_size The number of elements in the cache.
# TYPE coredns_cache_size gauge
coredns_cache_size{server="dns://0.0.0.0:53",type="denial"} 3
coredns_cache_size{server="dns://0.0.0.0:53",type="success"} 4
```

Reply:
I updated the gsg to check for `coredns_dns_request_count_total` instead as that directly reflects how many requests get redirected to nodelocal dns pod, the coredns_cache_{hits,misses}_total is internal to node-local-dns pod and might be different.